packages/plugin-session-replay/src/session-replay.ts

```diff
@@ -0,0 +1,286 @@
+import { AMPLITUDE_PREFIX, BaseTransport } from '@amplitude/analytics-core';
+import { BrowserConfig, EnrichmentPlugin, Event, PluginType, Status } from '@amplitude/analytics-types';
+import * as IDBKeyVal from 'idb-keyval';
+import { pack, record } from 'rrweb';
+import { shouldSplitEventsList } from './helpers';
+import { MAX_RETRIES_EXCEEDED_MESSAGE, STORAGE_FAILURE, SUCCESS_MESSAGE, UNEXPECTED_ERROR_MESSAGE } from './messages';
+import { Events, IDBStore, SessionReplayContext } from './typings/session-replay';
+
+const SESSION_REPLAY_SERVER_URL = 'https://api-secure.amplitude.com/sessions/track';
+const STORAGE_PREFIX = `${AMPLITUDE_PREFIX}_replay_unsent`;
+const PAYLOAD_ESTIMATED_SIZE_IN_BYTES_WITHOUT_EVENTS = 200; // derived by JSON stringifying an example payload without events
+const MAX_EVENT_LIST_SIZE_IN_BYTES = 20 * 1000000 - PAYLOAD_ESTIMATED_SIZE_IN_BYTES_WITHOUT_EVENTS;
+
+export class SessionReplayPlugin implements EnrichmentPlugin {
```

Comment:
only pointing this out because it's public facing. our other public facing plugins are written as functions and are camel cased (vs pascal cased). given it's a class i think it's understandable have it pascal-cased but for consistency with other public facing plugins, should we write this as a function and be camel cased? @amplitude/dx any thoughts on this?

Reply:
@kevinpagtakhan @justin-fiedler is this more about the name of the plugin or also about how it's written? Ie I can make it a function with a camel cased name by doing just
```
export const sessionReplayPlugin: SessionReplayPlugin = () => {
  return new SessionReplay();
};
```
Or are you asking the class also to be rewritten to match plugins like pageViewTracking? Asking before I go either direction because the changes have a large impact on my unit tests so hoping to avoid doing work that isn't what you're asking for